src/svm/web3-v2/solanaProgramUtils.ts

```diff
@@ -132,16 +162,89 @@ async function processEventFromTx(
  * For a given fillStatusPDa & associated spokePool ProgramID, return the fill event.
  */
 export async function readFillEventFromFillStatusPda(
-  rpc: web3.Rpc<web3.SolanaRpcApiFromTransport<RpcTransport>>,
+  client: RpcClient,
   fillStatusPda: Address,
   programId: Address,
   programIdl: Idl
 ): Promise<{ event: any; slot: number }> {
-  const signatures = await searchSignaturesUntilLimit(rpc, fillStatusPda);
+  const signatures = await searchSignaturesUntilLimit(client, fillStatusPda);
   if (signatures.length === 0) return { event: null, slot: 0 };
 
   // The first signature will always be PDA creation, and therefore CPI event carrying signature. Any older signatures
   // will therefore be either spam or PDA closure signatures and can be ignored when looking for the fill event.
-  const events = await readEvents(rpc, signatures[signatures.length - 1].signature, programId, programIdl);
+  const events = await readEvents(client, signatures[signatures.length - 1].signature, programId, programIdl);
   return { event: events[0], slot: Number(signatures[signatures.length - 1].slot) };
 }
+
+export async function createAlt(client: RpcClient, authority: KeyPairSigner): Promise<Address> {
+  const recentSlot = await client.rpc.getSlot({ commitment: "finalized" }).send();
+
+  const [alt] = await findAddressLookupTablePda({
+    authority: authority.address,
+    recentSlot,
+  });
+
+  const createAltIx = await getCreateLookupTableInstructionAsync({
+    authority,
+    recentSlot,
+  });
+
+  await pipe(
+    await createDefaultTransaction(client, authority),
+    (tx) => appendTransactionMessageInstruction(createAltIx, tx),
+    (tx) => signAndSendTransaction(client, tx)
+  );
+
+  return alt;
+}
+
+export async function extendAlt(client: RpcClient, authority: KeyPairSigner, alt: Address, addresses: Address[]) {
+  const extendAltIx = getExtendLookupTableInstruction({
+    address: alt,
+    authority,
+    payer: authority,
+    addresses,
+  });
+
+  await pipe(
+    await createDefaultTransaction(client, authority),
+    (tx) => appendTransactionMessageInstruction(extendAltIx, tx),
+    (tx) => signAndSendTransaction(client, tx)
+  );
+
+  const altAccount = await fetchAddressLookupTable(client.rpc, alt);
+
+  const addressesByLookupTableAddress: AddressesByLookupTableAddress = {};
+  addressesByLookupTableAddress[alt] = altAccount.data.addresses;
+
+  // Delay a second here to let lookup table warm up
+  await sleep(1000);
+
+  return addressesByLookupTableAddress;
+}
+
+async function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export async function sendTransactionWithAlt(
```

Comment:
in v1 we have `sendTransactionWithLookupTable` also creating and extending the ALT, is it possible to wrap this also in v2?